### PR TITLE
[Ace] Adds a feature fpic to enable -fPIC compilation on Linux

### PR DESCRIPTION
**Pull request by pierrebizz**
Adds a feature fpic to enable -fPIC compilation on Linux in order to use the ACE/TAO libraries in shared libs.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  <all>, <No>

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  `Yes`

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  <Yes>

### CODE REVIEW

File: ports/ace/portfile.cmake [hunk 1]
```diff
@@ -93,7 +93,7 @@ if(VCPKG_TARGET_IS_WINDOWS)
     set(SOLUTION_TYPE vc14)
   elseif(VCPKG_PLATFORM_TOOLSET MATCHES "v141")
     set(SOLUTION_TYPE vs2017)
-  elseif(VCPKG_PLATFORM_TOOLSET MATCHES "v142")
+  elseif(VCPKG_PLATFORM_TOOLSET MATCHES "v142")
```
Comment (JonLiu1993) [inline ports/ace/portfile.cmake:96]:
```suggestion
  elseif(VCPKG_PLATFORM_TOOLSET MATCHES "v142")
```

Comment (JonLiu1993) [reply ports/ace/portfile.cmake:96]: @pierrebizz ,Thanks for your pr, Please remove this extra space

Comment (pierrebizz) [reply ports/ace/portfile.cmake:96]: Done

**Comment by JonLiu1993** [comment]
@pierrebizz ,Please check your commit branch, it looks like you introduced a lot of unnecessary changes

**Comment by pierrebizz** [comment]
I will redo the Pull Request; I kind of rebased master instead of merging it ; I'm not familiar with GitHub workflow.

**Pull request closed without merging**